eclair-core/src/main/scala/fr/acinq/eclair/wire/ChannelCodecs.scala

```diff
@@ -92,9 +93,14 @@ object ChannelCodecs extends Logging {
     (wire: BitVector) => bool.decode(wire).map(_.map(b => if (b) IN else OUT))
   )
 
+  import shapeless.::
   val htlcCodec: Codec[DirectedHtlc] = (
     ("direction" | directionCodec) ::
-      ("add" | updateAddHtlcCodec)).as[DirectedHtlc]
+      ("add" | updateAddHtlcCodec)).xmap(
```

Comment:
Can't you get the result of the `directionCodec` here? I think we have some codecs that do that, I'll look around.

Reply:
Told you it had the potential to be rabbit hole-y 🤣

Reply:
Actually your proposal works fine, I like it. I just find the indentation a bit confusing, what about:

```scala
val htlcCodec: Codec[DirectedHtlc] = (("direction" | directionCodec) :: ("add" | updateAddHtlcCodec)).xmap(
    {
      case IN :: add :: HNil => IncomingHtlc(add)
      case OUT :: add :: HNil => OutgoingHtlc(add)
    },
    {
      htlc => htlc.direction :: htlc.add :: HNil
    }
  )
```

And we should put the `import shapeless.::` at the top of the file IMHO, not a big fan of scattering imports in the file.

Reply:
Will do, i also don't like the scattered imports but i'd make an exception for `shapeless.::` if imported at the top of the file it's very hard to understand what `::` does if it's from scala's collection or shapeless.